### PR TITLE
Rename cm.py to colormaps.py.

### LINKED ISSUE #1476: should cm be renamed to colormaps?
It's a bit confusing to have a proper module have such a short, non-descriptive name.

Looking at code that uses it, if one were to see `from matplotlib import cm` - they'd be left wondering what those two letters could possibly stand for. Then further down, they'd see it being used with a keyword argument that would shed some light, such as `cmap = cm.RdBu`, and they would maybe go through the steps of "oh, it's a map... probably a color map"...maybe...

If the import code was written as `from matplotlib import colormaps as cm`, it'd be more clear. We would retain the dangling `plt.cm` and have a dummy `matplotlib.cm` module which would just import everything from `matplotlib.colormaps`.

as a bonus, I think `plt.colormaps` should be retained, but the code and documentation for it should live inside this new colormaps module.

### PR DESCRIPTION
**Pull request by efiring**
Closes #1476.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by efiring** [comment]
This is the minimal version; actually changing all of the imports from `cm` to `colormaps` will touch a large number of files, and isn't urgent.  First, let's see whether there is support for doing this at all.

File: lib/matplotlib/cm.py [hunk 1]
```diff
@@ -1,389 +1,5 @@
 """
-Builtin colormaps, colormap handling utilities, and the `ScalarMappable` mixin.
+This is a compatibility shim for the renamed colormaps module.
```
Comment (timhoffm) [inline lib/matplotlib/cm.py]: Link to the colormaps module

**Comment by ImportanceOfBeingErnest** [comment]
Currently the `cm` module contains the public function `register_cmap`, `get_cmap` and the `ScalarMappable` class.  
The `colors` module contains a lot of color conversion stuff but also hosts the `Colormap` class and its descendents, as well as all the normalizations.  
So the confusing part would be that a new `colormaps.py` would neither contain the `Colormap`, nor the normalizations.

In total I'm a bit sceptical about replacing one confusion ("What does `cm` stand for?") with another one ("Why can't I import `Colormap` from `colormaps`?").

If truely rearanging stuff for the better, one would rather move the colormaps and normalizations out of `colors` and into `cm` - then renaming that module to  `colormaps` would make more sense, I'd say.

**Comment by ImportanceOfBeingErnest** [comment]
I just saw that [MEP21](https://matplotlib.org/devel/MEP/MEP21.html) says about `cm`

>cm
rename the module to something more descriptive - mappables?

**Comment by timhoffm** [comment]
I propose to hold this up until after the 3.0 release. If we are not absolutely sure this is the the right name, or if we want to do additional changes, that should be within one release. It would be really unfortunate to move stuff to *colormaps* in 3.0 and then to *mappables* in 3.1.

**Comment by jklymak** [comment]
Big :-1: on "mappables" :wink:.  That'd be far worse than "cm" in my opinion!

**Comment by tacaswell** [comment]
After a bit more thought I think pushing this to 3.1 so we can do the bigger re-organization, change all the examples, etc is the right thing to do.

**Comment by jklymak** [comment]
Re milestoning, but if we want to do this, we should try and actually do it :wink:

**Comment by efiring** [comment]
Support has evaporated...

**Comment by tacaswell** [comment]
We discussed this on the call and agreed that this will probably cause more trouble that it was worth.

**Pull request closed without merging**